File: COMP5216/ToDoList-Java-W1/app/src/main/java/comp5216/sydney/edu/au/todolist/MainActivity.java
package comp5216.sydney.edu.au.todolist;

import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;

public class MainActivity extends AppCompatActivity {
    // Define variables
    ListView listView;
    ArrayList<String> items;
    ArrayAdapter<String> itemsAdapter;
    EditText addItemEditText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // Use "activity_main.xml" as the layout
        setContentView(R.layout.activity_main);
        // Reference the "listView" variable to the id "lstView" in the layout
        listView = findViewById(R.id.lstView);
        addItemEditText = findViewById(R.id.txtNewItem);
        // Create an ArrayList of String
        items = new ArrayList<String>();
        items.add("item one");
        items.add("item two");
        // Create an adapter for the list view using Android's built-in item layout
        itemsAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, items);
        // Connect the listView and the adapter
        listView.setAdapter(itemsAdapter);
    }

    public void onAddItemClick(View view) {
        String toAddString = addItemEditText.getText().toString();
        Date now = new Date(System.currentTimeMillis());
        SimpleDateFormat format1 = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        String currentTime = format1.format(now);

        Toast.makeText(this, "Added this at " + currentTime, Toast.LENGTH_SHORT).show();
        if (toAddString.length() > 0) {
            itemsAdapter.add(toAddString + "  " + currentTime); // Add text to list view adapter
            addItemEditText.setText("");
        }
    }
}

File: COMP5216/ToDoList-Kotlin-W1/app/src/main/java/comp5216/sydney/edu/au/todolist_kotlin_w1/MainActivity.kt
package comp5216.sydney.edu.au.todolist_kotlin_w1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.EditText
import android.widget.ListView

class MainActivity : AppCompatActivity() {

    var listView: ListView? = null
    var items: ArrayList<String?>? = null
    var itemsAdapter: ArrayAdapter<String?>? = null
    var addItemEditText: EditText? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        listView = findViewById<View?>(R.id.lstView) as ListView
        addItemEditText = findViewById<View?>(R.id.txtNewItem) as EditText

        items = ArrayList()
        items!!.add("item one")
        items!!.add("item two")

        itemsAdapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, items!!)

        listView!!.adapter = itemsAdapter
    }

    fun onAddItemCLick(view: View?) {
        val toAddString = addItemEditText?.text.toString()
        if (toAddString.isNotEmpty()) {
            itemsAdapter?.add(toAddString)
            addItemEditText?.setText("")
        }
    }
}